bootstrap/bootstrap.go

```diff
@@ -59,12 +58,50 @@ const (
 	apiTypeSignin  = "signin"
 )
 
-// bootstrap is a data structure to hold configuration required to start
+// Defaults.
+const (
+	DefaultIdentifierClientPath = "./identifier-webapp"
+	DefaultSigningKeyID         = "default"
+	DefaultSigningKeyBits       = 2048
+)
+
+// Config is a typed application config which represents the user accessible config params
+type Config struct {
+	Iss                            string
+	IdentityManager                string
+	URIBasePath                    string
+	SignInURI                      string
+	SignedOutURI                   string
+	AuthorizationEndpointURI       string
+	EndsessionEndpointURI          string
+	Insecure                       bool
+	TrustedProxy                   []string
+	AllowScope                     []string
+	AllowClientGuests              bool
+	AllowDynamicClientRegistration bool
+	EncryptionSecretFile           string
+	Listen                         string
+	IdentifierClientPath           string
+	IdentifierRegistrationConf     string
+	IdentifierScopesConf           string
+	SigningKid                     string
+	SigningMethod                  string
+	SigningPrivateKeyFiles         []string
+	ValidationKeysPath             string
+	CookieBackendURI               string
+	CookieNames                    []string
+}
+
+// Bootstrap is a data structure to hold configuration required to start
 // konnectd.
-type bootstrap struct {
-	cmd  *cobra.Command
-	args []string
+type Bootstrap interface {
+	Config() *config.Config
+	Managers() *managers.Managers
+}
 
+// Bootstrap is a data structure to hold configuration required to start
```

Comment:
This doc string is not required and it does not say much about the actual function. Maybe say something about that this implements the Bootstrap interface?